repair/repair.cc

```diff
@@ -1280,7 +1284,7 @@ static future<> do_repair_ranges(lw_shared_ptr<repair_info> ri) {
             ri->ranges_index++;
             rlogger.info("Repair {} out of {} ranges, id={}, shard={}, keyspace={}, table={}, range={}",
                 ri->ranges_index, ri->ranges.size(), ri->id, ri->shard, ri->keyspace, ri->cfs, range);
-            return do_with(dht::selective_token_range_sharder(range, ri->shard), [ri] (auto& sharder) {
+            return do_with(dht::selective_token_range_sharder(ri->partitioner, range, ri->shard), [ri] (auto& sharder) {
                 return repeat([ri, &sharder] () {
```

Comment:
This function repairs multiple tables, so how can it accept a single partitioner?

Either it should use the partitioners from the tables, or it should verify that the table partitioners match the input partitioner.

Reply:
Actually, it does verify!